web-client/src/presenter/computeds/docketRecordHelper.test.js

```diff
@@ -0,0 +1,157 @@
+import { User } from '../../../../shared/src/business/entities/User';
+import { docketRecordHelper } from './docketRecordHelper';
+import { getUserPermissions } from '../../../../shared/src/authorization/getUserPermissions';
+import { runCompute } from 'cerebral/test';
+
+const getBaseState = user => {
+  return {
+    permissions: getUserPermissions(user),
+  };
+};
+
+describe('docket record helper', () => {
+  it('should show direct download link and not document detail link if the user is a petitioner', () => {
+    const user = {
+      role: User.ROLES.petitioner,
+      userId: '789',
+    };
+    const result = runCompute(docketRecordHelper, {
+      state: {
+        ...getBaseState(user),
+        caseDetail: {},
+        form: {},
+      },
+    });
+    expect(result.showDirectDownloadLink).toEqual(true);
+    expect(result.showDocumentDetailLink).toEqual(false);
+  });
+
+  it('should show direct download link and not document detail link if the user is a practitioner', () => {
+    const user = {
+      role: User.ROLES.practitioner,
+      userId: '789',
+    };
+    const result = runCompute(docketRecordHelper, {
+      state: {
+        ...getBaseState(user),
+        caseDetail: {},
+        form: {},
+      },
+    });
+    expect(result.showDirectDownloadLink).toEqual(true);
+    expect(result.showDocumentDetailLink).toEqual(false);
+  });
+
+  it('should show direct download link and not document detail link if the user is a respondent', () => {
+    const user = {
+      role: User.ROLES.respondent,
+      userId: '789',
+    };
+    const result = runCompute(docketRecordHelper, {
+      state: {
+        ...getBaseState(user),
+        caseDetail: {},
+        form: {},
+      },
+    });
+    expect(result.showDirectDownloadLink).toEqual(true);
+    expect(result.showDocumentDetailLink).toEqual(false);
+  });
+
+  it('should show document detail link and not direct download link if the user does have UPDATE_CASE permission', () => {
```

Comment:
@waldoj In this case, we're testing that the computed is returning what we expect based on the current user's role / application state. We don't need to directly test that the permission is being set, because that is handled in the `AUTHORIZATION_MAP`, which can be found here: https://github.com/ustaxcourt/ef-cms/blob/staging/shared/src/authorization/authorizationClientService.js

Reply:
This is weird. We're not testing whether the user has the `UPDATE_CASE` permission, and nowhere in the test do we establish that the permission exists. Were it not for the test description, there'd be no way to know that `UPDATE_CASE` was in play at all. If this test failed, I'd have a hard time understanding why, or what to do about it, because the thing that is being tested is not present within the test.

Am I making sense here?

Reply:
This has been addressed in my most recent commit. @waldoj let me know if that provided enough clarity or if we should add a comment somewhere as well.